### ISSUE TITLE
Add :discouraged-method linter to detect Java interop calls like System/exit

### ISSUE THREAD

**Issue by escherize** [clj-kondo/clj-kondo#2541]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**
Currently, there's no direct way to lint against Java interop method calls like `System/exit` in clj-kondo. I'm trying to prevent the use of `System/exit` in (parts of) our codebase, but the `:discouraged-var` linter doesn't detect it:

```shell
$ clj-kondo --version
clj-kondo v2025.04.07
$ echo '(ns hmm) (System/exit 12)' > hmm.clj
$ clj-kondo --lint hmm.clj --config '{:linters {:discouraged-var {java.lang.System/exit {:message "uh o" :level :warning}}}}'
linting took 4ms, errors: 0, warnings: 0
```

As borkdude confirmed in Slack, this functionality works for `analyze-call` but not for `discouraged-var` (referencing issue #1603), leaving us without a straightforward or built-in way to lint against these Java method calls. (Though it is possible with a custom hook).

**Describe the solution you'd like**
A new linter option, perhaps called `:discouraged-method` or `:discouraged-interop`, that works similarly to `:discouraged-var` but specifically targets Java interop method calls. This would allow configuration like:

```clojure
{:linters 
 {:discouraged-method 
  {java.lang.System/exit ;; <-- or just System/exit ?
   {:message "Please avoid System/exit as it bypasses normal shutdown hooks"
    :level :warning}}}}
```

The linter would detect when specified Java interop methods are called and report according to the configured level and message.

**Describe alternatives you've considered**
1. Writing a custom hook, but this seems like a generally useful feature that could benefit from being built into clj-kondo
2. Using regex-based approaches to scan for specific patterns like `System/exit`, but this is less reliable than proper syntactic analysis
3. Using `:analyze-call` directly, but see 1, I think it makes sense to not reinvent the wheel here.

**Additional context**
This would be valuable for enforcing coding standards that discourage certain Java interop calls that have better Clojure alternatives or that bypass important application lifecycle hooks (like `System/exit` vs throwing an ex-info with `{:babashka/exit 1}`).

The Slack conversation where borkdude confirmed this behavior can be found here: https://clojurians.slack.com/archives/CHY97NXE2/p1747759642038779
